pallets/ajuna-affiliate/src/tests.rs

```diff
@@ -0,0 +1,251 @@
+
+			// Third step on the chain BOB <- ALICE <- CHARLIE <- DAVE
+			assert_ok!(<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+				&CHARLIE, &DAVE
+			));
+
+			System::assert_last_event(mock::RuntimeEvent::Affiliates(
+				crate::Event::AccountAffiliated { account: DAVE, to: CHARLIE },
+			));
+
+			assert_eq!(Affiliators::<Test>::get(CHARLIE).affiliates, 1);
+			assert_eq!(Affiliatees::<Test>::get(DAVE), Some(bounded_vec![CHARLIE, ALICE]));
+
+			// Fourth step on the chain BOB <- ALICE <- CHARLIE <- DAVE <- EDWARD
+			assert_ok!(<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+				&DAVE, &EDWARD
+			));
+
+			System::assert_last_event(mock::RuntimeEvent::Affiliates(
+				crate::Event::AccountAffiliated { account: EDWARD, to: DAVE },
+			));
+
+			assert_eq!(Affiliators::<Test>::get(DAVE).affiliates, 1);
+			assert_eq!(Affiliatees::<Test>::get(EDWARD), Some(bounded_vec![DAVE, CHARLIE]));
+		});
+	}
+
+	#[test]
+	fn affiliate_to_rejects_with_self_account() {
+		ExtBuilder::default().build().execute_with(|| {
+			assert_noop!(
+				<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+					&ALICE, &ALICE
+				),
+				Error::<Test>::CannotAffiliateSelf
+			);
+		});
+	}
+
+	#[test]
+	fn affiliate_to_rejects_if_account_is_affiliator() {
+		ExtBuilder::default().build().execute_with(|| {
+			let state = AffiliatorState { status: AffiliatableStatus::Affiliatable, affiliates: 0 };
+			Affiliators::<Test>::insert(BOB, state);
+			Affiliators::<Test>::insert(ALICE, state);
+
+			assert_ok!(<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+				&BOB, &ALICE
+			));
+
+			assert_noop!(
+				<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+					&ALICE, &BOB
+				),
+				Error::<Test>::CannotAffiliateExistingAffiliator
+			);
+		});
+	}
+
+	#[test]
+	fn affiliate_to_rejects_affiliating_to_more_than_one_account() {
+		ExtBuilder::default().build().execute_with(|| {
+			let state = AffiliatorState { status: AffiliatableStatus::Affiliatable, affiliates: 0 };
+			Affiliators::<Test>::insert(BOB, state);
+			Affiliators::<Test>::insert(CHARLIE, state);
+
+			assert_ok!(<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+				&BOB, &ALICE
+			));
+
+			assert_noop!(
+				<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+					&CHARLIE, &ALICE
+				),
+				Error::<Test>::CannotAffiliateAlreadyAffiliatedAccount
+			);
+		});
+	}
+
+	#[test]
+	fn affiliate_to_rejects_with_unaffiliable_account() {
+		ExtBuilder::default().build().execute_with(|| {
+			assert_noop!(
+				<Affiliates as AffiliateMutator<AccountIdOf<Test>>>::try_add_affiliate_to(
+					&BOB, &ALICE
+				),
+				Error::<Test>::CannotAffiliateToAccount
+			);
+		});
+	}
+}
+
+mod clear_affiliation {
+	use super::*;
+
+	#[test]
+	fn clear_affiliation_should_work() {
```

Comment:
Maybe we should test `clear_affiliation_rejects_if_no_affiliation_exists`.

Reply:
Hmm currently clearing your affiliation if you have none will not cause any error, but I can add a test with the current behaviour for now!